tests/provider_upgrade_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2016-2023, Pulumi Corporation.  All rights reserved.
+
+package tests
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+
+	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+)
+
+const (
+	providerBinary = "pulumi-resource-aws"
+)
+
+var (
+	accept bool = os.Getenv("PULUMI_ACCEPT") != ""
```

Comment:
I know this is common, but sometimes I run `PULUMI_ACCEPT=false` and then it gets really confusing. 

```go
import 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
``` 
```suggestion
	accept bool = cmdutils.IsTruthy(os.Getenv("PULUMI_ACCEPT"))
```